cpp/tests/io/text/multibyte_split_test.cpp

```diff
@@ -62,6 +75,22 @@ TEST_F(MultibyteSplitTest, DelimiterAtEnd)
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, *out);
 }
 
+TEST_F(MultibyteSplitTest, DelimiterAtEndByteRange)
+{
+  auto delimiter  = std::string(":");
+  auto host_input = std::string("abcdefg:");
+
+  auto expected = strings_column_wrapper{"abcdefg:", ""};
+
+  auto source = cudf::io::text::make_source(host_input);
+  auto out    = cudf::io::text::multibyte_split(
+    *source,
+    delimiter,
+    cudf::io::text::byte_range_info{0, static_cast<int64_t>(host_input.size())});
```

Comment:
Thanks for making this test, it didn't occured to me we weren't testing this case explicitly, since under-the-hood we're using intmax as the byte range end when the byte range isn't specified.